src/assert.rs

```diff
@@ -80,6 +94,27 @@ impl Assert {
         }
     }
 
+    /// Run a specific example of the current crate.
+    ///
+    /// Defaults to asserting _successful_ execution.
+    pub fn example<S: AsRef<OsStr>>(name: S) -> Self {
```

Comment:
From commit message
> 
The `example` constructor is added to the `Assert` struct as a helper
method to run an example from the crate instead of having to use the
`command` constructor. This is similar to the `cargo_binary` but uses
the `--example` option instead of the `--bin` option.

Could you split this out into a dedicated PR so we can address these topics separately?